### PR TITLE
added new 2024 compliance data

### PR DESCRIPTION
**Pull request by freyac**
- added new compliance data that was published in Dec. 2024
- cleaned up python scripts to better handle year-to-year variations in CARB excel file processing
- fixed up the QR/QC `sandbox` notebook
- updated documentation to reflect new data

### CODE REVIEW

**Comment by freyac** [comment]
@badgley, would you be willing to do a spot check QR/QC on the updated data? 

@katamartin, this branch does not yet include necessary changes to 'layout.js', '.zenodo.json', or the front-end (new compliance period to filters, changing pointer to v4 of the data).

**Review by badgley — COMMENTED** [review]
quick skim — looks pretty good. i'll spot check data once its wired in vercel preview!

File: components/use-store.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import create from 'zustand'
 
 const url =
-  'https://raw.githubusercontent.com/carbonplan/compliance-users/main/data/outputs/user_data_v3.0.json'
+  'https://raw.githubusercontent.com/carbonplan/compliance-users/main/data/outputs/user_data_v4.0.json'
```
Comment (katamartin) [inline components/use-store.js:4]: I temporarily pointed this at `https://raw.githubusercontent.com/carbonplan/compliance-users/89408702a5c3c2e9b67fae954fb6b03acd4b4152/data/outputs/user_data_v4.0.json` and it looked good locally to me!

**Review by katamartin — COMMENTED** [review]
I'm not sure I'm following what the changes to `zenodo.json` and `layout.js` would be -- is there still more to do there?

But assuming the tool looks good with the temporary data pointer, it seems safe to merge with the ~broken pointer in this branch! (It looks like that what we did last year https://github.com/carbonplan/compliance-users/pull/121#discussion_r1444779803)

File: components/use-store.js (hunk 1 above)
Comment (freyac) [reply components/use-store.js:4]: Thanks, Kata! I had forgotten about that pattern. (cc @badgley for previewing the the new data on vercel)

I just changed the date in `zenodo.js` and had already changed the dates in `layout.js`, so I think we're good on that front.

**Pull request merged**